### PR TITLE
Renaming factor base types

### PR DESCRIPTION
**Pull request by GearsAD**
Merge after https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/pull/537

### CODE REVIEW

File: src/ApproxConv.jl [hunk 1]
```diff
@@ -252,12 +252,12 @@ end
 """
     $(SIGNATURES)
 
-Do true and null hypothesis computations based on data structures prepared earlier -- specific to `FunctorPairwiseNH`.  This function will be merged into a standard case for `FunctorPairwise/Minimize` in the future.
+Do true and null hypothesis computations based on data structures prepared earlier -- specific to `AbstractRelativeFactorNH`.  This function will be merged into a standard case for `AbstractRelativeFactor/Minimize` in the future.
```
Comment (dehann) [inline src/ApproxConv.jl:255]: `FunctorPairwiseNH` is about to deprecated anyway, so we can leave these.

Comment (dehann) [reply src/ApproxConv.jl:255]: part of the work to close #237

**Comment by GearsAD** [comment]
Ah, I think that's used in RoME though?

**Comment by GearsAD** [comment]
I'll revert the RoME changes for NH

**Pull request merged**